core/header.h

```diff
@@ -377,8 +377,8 @@ namespace MR
 
 
 
-
-
+  // Can't be a static member function due to memory alignment requirements of vector<>
```

Comment:
This is an odd one... What's the issue here, exactly?

Reply:
If I try to make that function a `static` member of `Header`, I get a compiler error, because it receives a `vector<Header>`, `vector<Header>` requires definition of aligned memory allocation functions as per `core/types.h`, but such functions can't be defined if the definition of class `Header` is not yet complete and hence its memory requirements are not yet confirmed.